docs/configuredgraphfactory.txt

```diff
@@ -0,0 +1,466 @@
+....
+
+Assuming the GremlinServer started successfully and the
+`JanusConfigurationGraph` was successfully instantiated, then all the
+APIs available on the `ConfigurationGraphManagement` `Singleton` will
+also act upon said graph. Furthermore, this is the graph that will be
+used to access the configurations used to create/open graphs using the
+`ConfiguredGraphFactory`.
+
+[[configurationgraphmanagement]]
+ConfigurationGraphManagement
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The `ConfigurationGraphManagement` is a `Singleton` that allows you to
+create/update/remove configurations that you can use to access your
+graphs using the `ConfiguredGraphFactory`. See above on configuring your
+server to enable use of these APIs.
+
+[[graph-configurations]]
+Graph Configurations
+^^^^^^^^^^^^^^^^^^^^
+
+The `ConfigurationGraphManagement` singleton allows you to create
+configurations used to open specific graphs, referenced by the
+"graph.graphname" property. For example:
+
+....
+Map<String, Object> map = new HashMap<String, Object>();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+map.put("graph.graphname", "graph1");
+ConfigurationGraphManagement.getInstance().createConfiguration(new MapConfiguration(map));
+....
+
+Then you could access this graph on any JanusGraph node using:
+
+....
+ConfiguredGraphFactory.open("graph1");
+....
+
+[[template-configuration]]
+Template Configuration
+^^^^^^^^^^^^^^^^^^^^^^
+
+The `ConfigurationGraphManagement` also allows you to create one
+template configuration, which you can use to create many graphs using
+the same configuration template. For example:
+
+....
+Map<String, Object> map = new HashMap<String, Object>();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+ConfigurationGraphManagement.getInstance().createTemplateConfiguration(new MapConfiguration(map));
+....
+
+After doing this, you can create graphs using the template
+configuration:
+
+....
+ConfiguredGraphFactory.create("graph2");
+....
+
+This method will first create a new configuration for "graph2" by
+copying over all the properties associated with the template
+configuration and storing it on a configuration for this specific graph.
+This means that this graph can be accessed in, on any JanusGraph node,
+in the future by doing:
+
+....
+ConfiguredGraphFactory.open("graph2");
+....
+
+[[updating-configurations]]
+Updating Configurations
+^^^^^^^^^^^^^^^^^^^^^^^
+
+[[important]]
+IMPORTANT
+^^^^^^^^^
+
+All interactions with both the `JanusGraphFactory` and the
+`ConfiguredGraphFactory` that interact with configurations that define
+the property "graph.graphname" go through the `JanusGraphManager` which
+keeps track of graph references created on the given JVM. Think of it as
+a graph cache. For this reason:
+
+_ANY UPDATES TO A CONFIGURATION ARE NOT GUARANTEED TO TAKE EFFECT UNTIL
+YOU REMOVE THE GRAPH IN QUESTION ON EVERY JANUSGRAPH NODE IN YOUR
+CLUSTER._
```

Comment:
no need to shout ;) If you precede the line with `IMPORTANT:` the docs generator will create a [call out box](http://docs.janusgraph.org/latest/server.html#_connecting_to_janusgraph_via_gremlin_server).